src/Nethermind/Nethermind.Evm/Tracing/CancellationTxTracer.cs

```diff
@@ -0,0 +1,394 @@
+            _token.ThrowIfCancellationRequested();
+            if (_innerTracer.IsTracingState)
+            {
+                _innerTracer.ReportStorageRead(storageCell);
+            }
+        }
+
+        public void MarkAsSuccess(Address recipient, long gasSpent, byte[] output, LogEntry[] logs, Keccak stateRoot = null)
+        {
+            _token.ThrowIfCancellationRequested();
+            if (_innerTracer.IsTracingReceipt)
+            {
+                _innerTracer.MarkAsSuccess(recipient, gasSpent, output, logs, stateRoot);
+            }
+        }
+
+        public void MarkAsFailed(Address recipient, long gasSpent, byte[] output, string error, Keccak stateRoot = null)
+        {
+            _token.ThrowIfCancellationRequested();
+            if (_innerTracer.IsTracingReceipt)
+            {
+                _innerTracer.MarkAsFailed(recipient, gasSpent, output, error, stateRoot);
+            }
+        }
+
+        public void StartOperation(int depth, long gas, Instruction opcode, int pc)
+        {
+            _token.ThrowIfCancellationRequested();
+            if (_innerTracer.IsTracingInstructions)
+            {
+                _innerTracer.StartOperation(depth, gas, opcode, pc);
+            }
+        }
+
+        public void ReportOperationError(EvmExceptionType error)
+        {
+            _token.ThrowIfCancellationRequested();
+            if (_innerTracer.IsTracingInstructions)
+            {
+                _innerTracer.ReportOperationError(error);
+            }
+        }
+
+        public void ReportOperationRemainingGas(long gas)
+        {
+            _token.ThrowIfCancellationRequested();
+            if (_innerTracer.IsTracingInstructions)
+            {
+                _innerTracer.ReportOperationRemainingGas(gas);
+            }
+        }
+
+        public void SetOperationStack(List<string> stackTrace)
+        {
+            _token.ThrowIfCancellationRequested();
+            if (_innerTracer.IsTracingStack)
+            {
+                _innerTracer.SetOperationStack(stackTrace);
+            }
+        }
+
+        public void ReportStackPush(in Span<byte> stackItem)
+        {
+            _token.ThrowIfCancellationRequested();
+            if (_innerTracer.IsTracingInstructions)
+            {
+                _innerTracer.ReportStackPush(stackItem);
+            }
+        }
+
+        public void ReportStackPush(in ZeroPaddedSpan stackItem)
+        {
+            _token.ThrowIfCancellationRequested();
+            if (_innerTracer.IsTracingInstructions)
+            {
+                _innerTracer.ReportStackPush(stackItem);
+            }
+        }
+
+        public void ReportStackPush(byte stackItem)
+        {
+            _token.ThrowIfCancellationRequested();
+            if (_innerTracer.IsTracingInstructions)
+            {
+                _innerTracer.ReportStackPush(stackItem);
+            }
+        }
+
+        public void SetOperationMemory(List<string> memoryTrace)
```

Comment:
Hmmm, did we introduce this tracer in new places? This is and stack are particularly expensive.

Reply:
This is disabled by default. Only an option for complitness.